### PR TITLE
Add embedError convenience function

### PR DESCRIPTION
**Pull request by ChristopherDavenport**
Not much more to it. Just a convenience function. 

Especially useful when you are trying to save the outcome of some computation. But then you need to resume flow.

### CODE REVIEW

File: kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala [hunk 1]
```diff
@@ -162,6 +162,15 @@ object Outcome extends LowPriorityImplicits {
...
     either.fold(Errored(_), a => Succeeded(a.pure[F]))
 
+  def embedError[F[_], A](outcome: Outcome[F, Throwable, A])(
+      implicit F: ApplicativeError[F, Throwable]): F[A] =
+    outcome match {
+      case Succeeded(fa) => fa
+      case Errored(e) => F.raiseError(e)
+      case Canceled() =>
+        F.raiseError(new java.util.concurrent.CancellationException("Outcome was Canceled"))
+    }
```
Comment (vasilmkd) [inline kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala]: `Outcome` has a `fold` method for this pattern match.

**Review by vasilmkd — DISMISSED** [review]

**Review by vasilmkd — COMMENTED** [review]
I think it's worth bikeshedding on the name and adding an accompanying scaladoc.

File: kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala (hunk 1 above)
Comment (ChristopherDavenport) [reply kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala]: I switched to embed which seems more direct. Can use fold if you'd prefer.

Comment (vasilmkd) [reply kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala]: Even better.

**Comment by vasilmkd** [comment]
I think the scaladoc should mention the cancelation case, saying that it embeds cancelation as an error. Otherwise 👍🏻

File: kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala [hunk 2]
```diff
@@ -166,7 +166,8 @@ object Outcome extends LowPriorityImplicits {
    * embedError allows the restoration to a normal development flow from an Outcome.
    *
    * This can be useful for storing the state of a running computation and then waiters for that
-   * data can act and continue forward on that shared outcome.
+   * data can act and continue forward on that shared outcome. Cancelation is encoded
+   * as an exception.
```
Comment (armanbilge) [inline kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala]:
```suggestion
   * as a `CancellationException`.
```

File: kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala [hunk 3]
```diff
@@ -162,6 +162,19 @@ object Outcome extends LowPriorityImplicits {
...
     either.fold(Errored(_), a => Succeeded(a.pure[F]))
 
+  /**
+   * embedError allows the restoration to a normal development flow from an Outcome.
+   *
+   * This can be useful for storing the state of a running computation and then waiters for that
+   * data can act and continue forward on that shared outcome. Cancelation is encoded as a
+   * `CancellationException`.
+   */
+  def embedError[F[_], A](outcome: Outcome[F, Throwable, A])(
```
Comment (djspiewak) [inline kernel/shared/src/main/scala/cats/effect/kernel/Outcome.scala]: I think this should be a member of `Outcome` like the other `embed`s.

**Review by djspiewak — CHANGES_REQUESTED** [review]

**Review by djspiewak — APPROVED** [review]

**Pull request merged**